### PR TITLE
GitHub issue template for new features and bugs

### PR DESCRIPTION
**Pull request by deeleman**
This PR summarizes what @paoloantinori, @rhuss and me discussed over the F2F regarding having a template model for new GH issues that would serve as a centralized point of reference for issues/bugs combining UI/API/Backend input.

The proposed template tries to cater with both bugs and new features. GitHub provides support for handling several issue templates in a single repository. However, I'd suggest to begin with this one and refine its content. In the future we can provide more than one template (new feature, bugs, etc), which we can trigger by means of custom browser script bookmarks (or even a custom browser extension).

### CODE REVIEW

**Comment by zregvart** [comment]
Thanks! I'm reading the [docs](https://help.github.com/articles/creating-an-issue-template-for-your-repository/) and the example there uses lowercase filename (`issue_template.md`), is the support for this case-insensitive?

I would perhaps use headings level3 (`###` instead of `##`)?

I think we need to prompt for version or environment details (if relevant). Do we need a reminder to try with the latest version also?

**Comment by deeleman** [comment]
> is the support for this case-insensitive?

I presume it is. I just checked the [Angular setup](https://github.com/angular/angular/tree/master/.github) for this and it uses uppercase, which is BTW quite a standardized convention for markdown files in GitHub repos anyways. [Some other repos](https://github.com/stevemao/github-issue-templates) I've checked seem to follow the same convention.

> I think we need to prompt for version or environment details (if relevant). 

Agreed. It is actually quite a standardized practice. I'm limited here to my little knowledge on the matter though (I'm familiar with browser environments but not with the other environments that might impact the backed or the execution context) so commits and additional PRs are welcome here. 

Ultimately this PR is intended to be a group endeavor, so feel free to edit the template to add whatever you feel is necessary. :)

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+## This is a...
+<!-- Check one of the following options with "x" -->
+<pre><code>
```
Comment (rhuss) [inline .github/ISSUE_TEMPLATE.md]: when we leave out the preformatting,then it should be directly rendered as checkboxes by GitHub Markdown, like in

- [x] Finish my changes
- [ ] Push my commits to GitHub
- [ ] Open a pull request

**Comment by rhuss** [comment]
Looks good to me, and this templates is supposed to be adapted anyway when needed, so I suggest to merge it.

**Comment by paoloantinori** [comment]
@deeleman question for you:

is this meant to be used for bugs or even for greenfield development?

could you use it for you Log story for example, or should we reason on something different for that?

**Pull request merged**